src/metabase/lib/types/isa.clj

```diff
@@ -0,0 +1,95 @@
+(ns metabase.lib.types.isa
+  "Ported from frontend/src/metabase-lib/types/utils/isa.js.
+
+  This code belong to MLv2 project and it belongs to 47,
+  we only ported some functionality that is needed for a bug fix in 46.
+  See #30321 for the original PR."
+  (:require
+   [medley.core :as m]
+   [metabase.types])
+  (:refer-clojure :exclude [isa? any? boolean? number? string?]))
+
+(comment metabase.types/keep-me)
+
+(def type-hierarchies
```

Comment:
this is actually from https://github.com/metabase/metabase/blob/b54802111ea901fc81e8d583e04d7427428443d9/src/metabase/lib/types/constants.cljc but I copied it here because I don't want to create another ns just for it.